### DISCUSSION TITLE
Document Head nonce support doesn't work in development

### DISCUSSION THREAD

**Discussion by ericbiewener** [vercel/next.js#32227] [Ideas]
### What version of Next.js are you using?

10.1.3

### What version of Node.js are you using?

14.15.1

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

Other

### Describe the Bug

[This issue](https://github.com/vercel/next.js/issues/18557) was supposedly fixed by [this PR](https://github.com/vercel/next.js/pull/19150), but I'm not seeing the `nonce` applied to `<style>` tags when running in development.

### Expected Behavior

`<style>` tags generated by Next include the `nonce`, even in development.

### To Reproduce

This repo contains a minimal reproduction: https://github.com/ericbiewener/next-nonce-bug

**Comment by jcdogo** [comment]
It looks like this is not working in development because of these lines:

https://github.com/vercel/next.js/blob/master/packages/next/client/index.tsx#L664-L671

The code checks if the server is running in development mode and the nonce fix never runs if in development mode.

@styfle Since you were the one that implemented the fix, can you comment on how to get this working in development mode?

**Comment by styfle** [comment]
Generally speaking, CSP isn't necessary for `next dev`. The `if` statement you linked to was added in PR #16126 because CSS is inlined during dev and CSP would block it.

**Comment by ericbiewener** [comment]
I don't understand. If the CSP has a `styles-src nonce-abc123` policy, then the inline `<style>` tags need to support the nonce attribute. Therefore, in order to prevent my styles from being blocke din dev, I have to use `unsafe-inline` in the CSP when in dev.
